src/casp_cli/service.py

```diff
@@ -0,0 +1,109 @@
+        return f"{cls.BACKEND_URL}/{endpoint}"
+
+    async def post(self, endpoint: str, data) -> t.Dict:
+        url = self._get_endpoint_url(endpoint=endpoint)
+
+        async with aiohttp.ClientSession() as session:
+            async with session.post(url, data=data) as resp:
+                return await resp.json()
+
+
+class CASPClientService(_BaseService):
+    BACKEND_URL = "https://cas-manager-vi7nxpvk7a-uc.a.run.app"
+
+    def _get_number_of_chunks(self, adata, chunk_size):
+        return math.ceil(len(adata) / chunk_size)
+
+    def _get_timestamp(self) -> str:
+        return datetime.datetime.now().strftime("%H:%M:%S.%f")[:-3]
+
+    def _print(self, str_to_print: str) -> None:
+        print(f"* [{self._get_timestamp()}] {str_to_print}")
+
+    async def _annotate_anndata_chunk(
+        self,
+        adata_bytes: bytes,
+        results: t.List,
+        chunk_index: int,
+        chunk_start_i: int,
+        chunk_end_i: int,
+        start_time: float,
+    ) -> None:
+        data = {"myfile": adata_bytes}
+        for _ in range(3):
+            try:
+                results[chunk_index] = await self.post("annotate", data=data)
+            except Exception:
+                self._print(f"Something went wrong, resubmitting chunk #{chunk_index + 1:2.0f} ({chunk_start_i:5.0f}, {chunk_end_i:5.0f}) to CAS ...")
+                pass
+            else:
+                break
+        self._print(f"Received the annotations for cell chunk #{chunk_index + 1:2.0f} ({chunk_start_i:5.0f}, {chunk_end_i:5.0f}) ...")
+
+    async def _annotate_anndata_task(self, adata, results, chunk_size, start_time) -> None:
+        i, j = 0, chunk_size
+        tasks = []
+        number_of_chunks = self._get_number_of_chunks(adata, chunk_size=chunk_size)
+        for chunk_index in range(number_of_chunks):
+            chunk = adata[i:j, :]
+            chunk_start_i = i
+            chunk_end_i = i + len(chunk)
+            self._print(f"Submitting cell chunk #{chunk_index + 1:2.0f} ({chunk_start_i:5.0f}, {chunk_end_i:5.0f}) to CAS ...")
+            tmp_file_name = f"chunk_{chunk_index}.h5ad"
+            chunk.write(tmp_file_name, compression="gzip")
+            with open(tmp_file_name, "rb") as f:
+                chunk_bytes = f.read()
+
+            os.remove(tmp_file_name)
+            tasks.append(
+                self._annotate_anndata_chunk(
+                    adata_bytes=chunk_bytes,
+                    results=results,
+                    chunk_index=chunk_index,
+                    chunk_start_i=chunk_start_i,
+                    chunk_end_i=chunk_end_i,
+                    start_time=start_time,
+                )
+            )
+            i = j
+            j += chunk_size
+
+        await asyncio.wait(tasks)
+
+    def annotate_anndata(self, adata: "anndata.AnnData", chunk_size=2000) -> t.List:
+        start = time.time()
+        self._print("CAS v1.0 (Model ID: PCA_002)")
```

Comment:
call a function to get these, that could be static for now but really should come from the API somehow (or be parameters from the user as to which model they want to use).